nemo/src/io/formats/dsv.rs

```diff
@@ -387,7 +415,10 @@ impl FileFormatMeta for DSVFormat {
     }
 
     fn optional_attributes(&self, _direction: Direction) -> HashSet<Key> {
-        [].into()
+        match _direction {
+            Direction::Reading => [Key::identifier_from_str(MAP_COLUMNS)].into(),
+            Direction::Writing => [].into(),
```

Comment:
In theory it's much easiear, since there would be no reordering / projection. I'd name it something different though, to make that difference more visible (it is not really a "selection" or "projection" in this case). Maybe just `add_header` or `header`?

Reply:
how about `column_names` (since, fundamentally, that is what needs to be specified for writing)? That could even work for both reading and writing.